File: Components/UIComponents/Common/NewRowButton.py
# Set shebang if needed
# -*- coding: utf-8 -*-
"""
Created on Sat Jan 10 19:01:32 2026

@author: mano
"""

from dash import html
from Components.UIComponents.Managers.UIManager import UIManager


UIM = UIManager()
def NewRowButtonComp(name, row_lv1=None, disabled=False):

    boton = html.Button(
        children=[
            html.Span('+', className='plus-icon'),
            html.Span('Añadir Nuevo ' + name)
        ],
        n_clicks=0,
        **{'id': UIM.id_generator(ui_type='Input',
                                  ui_name=name,
                                  ui_subtype='Button',
                                  row_lv1=row_lv1)},
        className='btn-add-filter',
        disabled=disabled
    )
    comp = html.Div(
        children = [boton],
        className = 'add-filter-container'
    )
    return comp

File: assets/JavaScript/UI/Common/NewRowButton.mjs

import dash from './dash_components.mjs';
import id_generator from './Functions/id_generator.mjs';

function NewRowButtonComp(name, row_lv1=null, disabled=false){

    const boton = dash._html.Button({
        'children': [
            dash._html.Span({'children': '+', 'className':'plus-icon'}),
            dash._html.Span({'children': 'Añadir nuevo ' + String(name)})
        ],
        'n_clicks': 0,
        'id': id_generator('Input', name, 'Button', row_lv1),
        'className': 'btn-add-filter',
        'disabled': disabled
    })
    const comp = dash._html.Div({
        'children': [boton],
        'className': 'add-filter-container'
    })
    return comp;
}

export default NewRowButtonComp;